### PR TITLE
[5.0] Fix addSelect behavior

### PR DESCRIPTION
**Pull request by rkgrep**
Solution to this bug report: #6513

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -278,6 +278,11 @@ public function addSelect($column)
 	{
 		$column = is_array($column) ? $column : func_get_args();
 
+		if (is_null($this->columns))
+		{
+			$this->columns = [ '*' ];
+		}
+
 		$this->columns = array_merge((array) $this->columns, $column);
```
Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: Can we simplify this whole bit using the following:

```php
$this->columns = array_merge($this->columns ?: ['*'], $column);
```

Comment (rkgrep) [inline src/Illuminate/Database/Query/Builder.php]: There is difference between empty columns and null columns variable - first case appears if user has passed empty array to select for some reason.
```php
$model->select([])->addSelect('test');
```
Which is a different situation from
```php
$model->addSelect('foo');
```
Shortened code will look much less readable
```php
$this->columns = array_merge(is_null($this->columns) ? ['*'] : $this->columns, $column);
```

Comment (JoostK) [inline src/Illuminate/Database/Query/Builder.php]: No this is fine then. I didn't consider empty arrays as being useful. We could probably ditch the array cast in `array_merge` though, as we should now always have an array.

Do please remove the surrounding whitespace in `[ '*' ]` though :wink:

**Comment by rkgrep** [comment]
CI checks failed because of composer problems

**Comment by GrahamCampbell** [comment]
Retriggering.

**Pull request closed without merging**